src/main/java/io/github/enkarin/bookcrossing/user/dto/UserDetailsDto.java

```diff
@@ -0,0 +1,78 @@
+package io.github.enkarin.bookcrossing.user.dto;
+
+import io.github.enkarin.bookcrossing.user.model.User;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import javax.annotation.concurrent.Immutable;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Immutable
```

Comment:
This class is not immutable because it returns internal representation of `roles`

Reply:
It returns RoleDto